tests/testthat/test-cache.R

```diff
@@ -86,9 +86,13 @@ test_that("load_cache returns an empty environment if reading cache file fails",
   cache_f1 <- file.path(d1, fhash(f1))
   writeLines(character(), cache_f1)
 
-  expect_warning(e2 <- load_cache(file = f1, path = d1))
+  expect_warning(e2 <- load_cache(file = f1, path = d1), "Could not load cache file")
   saveRDS(e1, cache_f1)
-  expect_warning(e3 <- load_cache(file = f1, path = d1))
```

Comment:
> The concrete warnings are version-dependent.

a bit too vague as a comment (I'm imagining reading this comment in 4 years' time).

maybe a `skip()` for older versions here is appropriate?

Reply:
I've never worked with snapshots before. Do they support different R versions giving different output?

Reply:
Different suggestion to keep the code somewhat clean: Wrap it in `withCallingHandlers()` and explicitly check the bubbled warning messages for those that we know of, discarding them silently and bubbling up any additional warnings. WDYT?

Reply:
yes, via `variant=`.

See testthat's suite e.g. https://github.com/r-lib/testthat/tree/main/tests/testthat/_snaps.

Personally I think tidyverse started over-relying on them -- they're very fragile (e.g. an upstream like `crayon` can make a small change in how output is displayed and now your snapshot test is broken) and have been hell to maintain inside Google, and are turned off on CRAN by default.

But it could work to use them in limited cases. Something to consider at least.

Reply:
> bubbling up any additional warnings

I'll have to look at the implementation for what this means exactly. But SGTM anyway.

Reply:
Something like that: `withCallingHandlers({ warning("ok"); warning("oops") }, warning = function(w) {if (conditionMessage(w) == "ok") invokeRestart("muffleWarning") else w})`